src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.hpp

```diff
@@ -0,0 +1,182 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <limits>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/Systems/Elasticity/FirstOrderSystem.hpp"
+#include "Elliptic/Systems/Elasticity/Tags.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/IsotropicHomogeneous.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Elasticity::Solutions {
+/*!
+ * \brief The solution for a half-space mirror deformed by a laser beam.
+ *
+ * \details This solution is mapping thermal noise to an elasticity problem
+ * where a normally incident and axisymmetric laser beam with a Gaussian beam
+ * profile acts on the face of a semi-infinite mirror. Here we assume the face
+ * to be at \f$z = 0\f$ and the material to extend to \f$+\infty\f$ in the
+ * z-direction as well as for the mirror diameter to be comparetively large
+ * to the `beam width`. The mirror material is characterized by an isotropic
+ * homogeneous constitutive relation \f$Y^{ijkl}\f$ (see
+ * `Elasticity::ConstitutiveRelations::IsotropicHomogeneous`). In this scenario,
+ * the laser introduces a pressure in the form of a Gaussian distribution
```

Comment:
I would say instead "the auxiliary elastic problem has an applied pressure distribution equal to the laser beam intensity profile"